core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -511,6 +511,8 @@ protected Scope visitRefreshMaterializedView(RefreshMaterializedView refreshMate
                 throw semanticException(TABLE_NOT_FOUND, refreshMaterializedView, "Materialized view '%s' does not exist", name);
```

Comment:
Could you also add test like https://github.com/trinodb/trino/pull/7618/files#diff-7626bb40b118fe1c19273979307bee76926268ca477163c71e594378886554c2 
that checks that user is able to perform MV refresh without having access to storage table?

Reply:
added testRefreshAllowedWithRestrictedStorageTable

Reply:
nit: still, engine only test would be nice, but it's fine now